libcaf_net/caf/net/multiplexer.cpp

```diff
@@ -221,12 +221,14 @@ class default_multiplexer : public detail::atomic_ref_counted,
 
   // -- thread-safe signaling --------------------------------------------------
 
-  void start(socket_manager_ptr mgr) override {
+  bool start(socket_manager_ptr mgr) override {
     auto lg = log::net::trace("socket = {}", mgr->handle().id);
     if (std::this_thread::get_id() == tid_) {
       do_start(mgr);
+      return !shutting_down_;
```

Comment:
We already check the flag in `do_start`. Might as well return properly from that function.

```suggestion
      return do_start(mgr);
```